src/common/tracer.h

```diff
@@ -79,26 +80,25 @@ struct span_stub {
   bool IsRecording() { return false; }
 };
 
-class jspan {
-  span_stub span;
- public:
-  span_stub& operator*() { return span; }
-  const span_stub& operator*() const { return span; }
-
-  span_stub* operator->() { return &span; }
-  const span_stub* operator->() const { return &span; }
-
+struct jspan_ptr {           
+  jspan span;           
```

Comment:
is there a reason this needs to be public now?

Reply:
i think it should remain private if there's no reason to change it. but this doesn't need to delay qa